### PR TITLE
enable grad check

### PR DESCRIPTION
**Pull request by Krovatkin**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58249 Gelu Backward, Contribution from Kevin Stephano
* #57750 hardtanh
* #57749 init hardshrink
* **#57828 enable grad check**

Differential Revision: [D28425381](https://our.internmc.facebook.com/intern/diff/D28425381)

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -4782,13 +4785,14 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
     OpInfo('nn.functional.hardswish',
+           aten_name="hardswish",
            supports_autograd=True,
            assert_autodiffed=True,
            sample_inputs_func=sample_inputs_hardswish,
            dtypesIfCUDA=floating_types_and(torch.half, torch.bfloat16),
            supports_gradgrad=False,
            supports_out=False,
-           autodiff_fusible_nodes=["aten::hardswish"]),
+           autodiff_nonfusible_nodes=["aten::hardswish"]),
```
Comment (eellison) [inline torch/testing/_internal/common_methods_invocations.py:4795]: why was this change made ?

Comment (Krovatkin) [reply torch/testing/_internal/common_methods_invocations.py:4955]: fusible means it's inside a fusion group inside a differentiable graph and nonfusible means inside a differentiable graph

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3942,6 +3944,7 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypes=floating_types(),
            supports_out=False,
            supports_gradgrad=False,
+           assert_autodiffed=False,
```
[the remark this answers is not in the document]
Comment (Krovatkin) [reply torch/testing/_internal/common_methods_invocations.py:4059]: `cdist` doesn't have a symbolic backward, but we would at least like to run forward through torchscript.

File: test/test_ops.py [hunk 3]
```diff
@@ -362,9 +362,6 @@ def _test_inplace_preserve_storage(samples, variants):
     def test_variant_consistency_jit(self, device, dtype, op):
         _requires_grad = op.supports_autograd and (dtype.is_floating_point or
                                                    op.supports_complex_autograd(torch.device(device).type))
-        # TODO: fix this
-        if _requires_grad and not op.supports_gradgrad:
```
Comment (mruberry) [inline test/test_ops.py:366]: Wahoo!

File: test/test_ops.py [hunk 4]
```diff
@@ -404,7 +401,7 @@ def out_fn(output):
                                             out_fn,
                                             (sample.input,) + sample.args,
                                             sample.kwargs,
-                                            no_grad=not _requires_grad)
+                                            no_grad=not _requires_grad, no_gradgrad=not op.supports_gradgrad)
```
Comment (mruberry) [inline test/test_ops.py:404]: This is a cool extension

cc @Lilyjjo

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -1090,7 +1090,9 @@ def sample_inputs_cdist(op_info, device, dtype, requires_grad, **kwargs):
 
     samples = []
     for cm in ['use_mm_for_euclid_dist', 'donot_use_mm_for_euclid_dist']:
-        for p in [0, 1, 2, 3, 0.5, 1.5, 2.5, float("inf")]:
+        # FIXME add an override for JIT and revert 0. back to 0
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:1093]: Would you file an issue for this and link the issue, too?

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:4081]: It's awesome that this test is actually running now to detect these issues

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:4977]: @eellison the test wasn't running previously so I think @Krovatkin is just correcting the metadata

**Review by mruberry — APPROVED** [review]
The OpInfo-related changes look great! I made one suggestion to add a link to an issue in a comment to help us track the fix

File: torch/testing/_internal/common_methods_invocations.py (hunk 5 above)
Comment (Krovatkin) [reply torch/testing/_internal/common_methods_invocations.py:1093]: yes will do

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425381).

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425381).

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425381).

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425381).

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28425381).

**Pull request closed without merging**